### PR TITLE
Additional fixes for documentation in demos.

### PR DESCRIPTION
**Pull request by clalancette**
With these fixes, I think we should get a green Rdoc build.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

File: intra_process_demo/include/image_pipeline/watermark_node.hpp [hunk 1]
```diff
@@ -26,11 +26,16 @@
...
 
-// Node that receives an image, adds some text as a watermark, and publishes it again.
-class WatermarkNode : public rclcpp::Node
+/// Node that receives an image, adds some text as a watermark, and publishes it again.
+class WatermarkNode final : public rclcpp::Node
 {
 public:
-  WatermarkNode(
+  /// \brief Construct a WatermarkNode that accepts an image, adds a watermark, and republishes it
+  /// \param input The topic to listen on
```
Comment (hidmic) [inline intra_process_demo/include/image_pipeline/watermark_node.hpp:34]: @clalancette nit^2:
```suggestion
  /// \param input The name of the topic to subscribe to
```
We tend to use `topic` and `topic name` interchangeably elsewhere too, so it's not a strong opinion.

File: quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp [hunk 2]
```diff
@@ -97,18 +99,18 @@ class Talker : public rclcpp::Node
   rclcpp::TimerBase::SharedPtr assert_topic_timer_ = nullptr;
 };
 
-class Listener : public rclcpp::Node
+/// class to listen to listen to incoming data from the Talker
```
Comment (hidmic) [inline quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp:102]: @clalancette nit:
```suggestion
/// Node class to listen to incoming data from the Talker
```

File: quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp [hunk 3]
```diff
@@ -24,19 +24,21 @@
 
 constexpr char DEFAULT_TOPIC_NAME[] = "qos_chatter";
 
-class Talker : public rclcpp::Node
+/// Class to publish data at a specified rate, optionally pausing to test liveliness
```
Comment (hidmic) [inline quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp:27]: @clalancette nit: 
```suggestion
/// Node class to publish data at a specified rate, optionally pausing to test liveliness
```

File: intra_process_demo/include/image_pipeline/image_view_node.hpp [hunk 4]
```diff
@@ -24,10 +24,14 @@
...
 #include "common.hpp"
 
-// Node which receives sensor_msgs/Image messages and renders them using OpenCV.
-class ImageViewNode : public rclcpp::Node
+/// Node which receives sensor_msgs/Image messages and renders them using OpenCV.
+class ImageViewNode final : public rclcpp::Node
 {
 public:
+  /// \brief Construct a new ImageViewNode for visualizing image data
+  /// \param input The topic name to subscriber to
```
[the remark this answers is not in the document]
Comment (clalancette) [reply intra_process_demo/include/image_pipeline/image_view_node.hpp]: Fixed in 604bfcf

File: intra_process_demo/include/image_pipeline/watermark_node.hpp (hunk 1 above)
Comment (clalancette) [reply intra_process_demo/include/image_pipeline/watermark_node.hpp]: Fixed in 604bfcf

File: quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp (hunk 2 above)
Comment (clalancette) [reply quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp]: Fixed in 604bfcf

File: quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp (hunk 3 above)
Comment (clalancette) [reply quality_of_service_demo/rclcpp/include/quality_of_service_demo/common_nodes.hpp]: Fixed in 604bfcf

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15577/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10262/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13246/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15797/)

**Review by hidmic — APPROVED** [review]

**Comment by clalancette** [comment]
The yellow Windows build is unfortunate, but I have a hard time seeing how they could be related to this PR.  The failures are all in either `rclcpp` or in `rosbag2`, neither of which are affected by this PR.

I'm going to go ahead and assume they are unrelated failures and merge.  Thanks for the review!

**Pull request merged**